src/Interpreters/SystemLog.cpp

```diff
@@ -517,8 +517,7 @@ void SystemLog<LogElement>::flushImpl(const std::vector<LogElement> & to_flush,
         ASTPtr query_ptr(insert.release());
 
         // we need query context to do inserts to target table with MV containing subqueries or joins
-        auto insert_context = Context::createCopy(context);
-        insert_context->makeQueryContext();
+        auto insert_context = getQueryContext(getContext());
```

Comment:
To me it seems it would have been better to have some virtual method like `addSettings()` or `addSettingsIfNeeded` instead of `getQueryContext` e.g. to add some settings similar to how they are added just one line below: `insert_context->setSetting("materialized_views_ignore_errors", true);`. Because it will be more transparent for the method actually does (not it is visible only if you read a comment near `getQuerySettings` method)

Reply:
Added 

```
    virtual void addSettingsForQuery(ContextMutablePtr & mutable_context, IAST::QueryKind query_kind) const;
```

I was not sure if we need to move existing `setSetting` there, but did it